.github/workflows/publish-docs.yml

```diff
@@ -62,19 +63,56 @@ jobs:
         if: contains(github.ref, 'refs/heads/main')
         run: |
-          mike deploy --push main
+          mike delete main || true
+          mike deploy --push head
 
-      - name: Get the release version
-        if: contains(github.ref, 'refs/tags/v') && !github.event.release.prerelease # (don't generate for pre-releases)
-        id: get_version
-        run: echo ::set-output name=VERSION::${GITHUB_REF/refs\/tags\//}
+      # Lists tags using the "gh" cli
+      # Gets rid of old versions (ones without +k0s) and drafts using grep
+      # The list is sorted using k0s_sort installed "install dependencies"
+      - name: Collect tags
+        id: tags
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          TAGS=$(gh release list -L 1000 -R k0sproject/k0s | grep "+k0s." | grep -v Draft | cut -f 1 | k0s_sort)
+          echo ::set-output name=SORTED_TAGS::${TAGS}
 
-      - name: mike deploy new release
-        if: contains(github.ref, 'refs/tags/v') && !github.event.release.prerelease # (don't generate for pre-releases)
+      # If this is a tag build, deploy as a new version
+      - name: mike deploy new version
+        if: contains(github.ref, 'refs/tags/v') && !github.event.release.prerelease
         run: |
-          mike deploy --push --update-aliases ${{ steps.get_version.outputs.VERSION }} latest
-          mike set-default --push ${{ steps.get_version.outputs.VERSION }}
+          VERSION=${GITHUB_REF/refs\/tags\//}
+          mike deploy --push "${VERSION}"
 
+      - name: Update mike version aliases
+        id: set_versions
+        run: |
+          LATEST=$(echo ${{ steps.tags.outputs.SORTED_TAGS }} | tail -1)
+          STABLE=$(echo ${{ steps.tags.outputs.SORTED_TAGS }} | grep -v -- "-" | tail -1)
+          mike alias -u "${LATEST}" latest
+          mike alias -u "${STABLE}" stable
+          mike set-default --push stable
+          echo ::set-output name=LATEST::${LATEST}
+          echo ::set-output name=STABLE::${STABLE}
+
+      # Ensures the current branch is gh-pages,
+      # Creates / updates the "stable" and "latest" plain text files with the corresponding versions
+      # Commits if the files were changed
+      # Finally pushes if there are unpushed commits
+      - name: Create version files
+        run: |
+          LATEST=${{ steps.set_versions.outputs.LATEST }}
+          STABLE=${{ steps.set_versions.outputs.STABLE }}
+          (git branch --show-current | grep -q gh-pages) || git checkout gh-pages
+          echo "${STABLE}" > stable
+          git add stable && git update-index --refresh
+          git diff-index --quiet HEAD -- || git commit -m "Set stable to ${STABLE}"
+          echo "${LATEST}" > latest
+          git add latest && git update-index --refresh
+          git diff-index --quiet HEAD -- || git commit -m "Set latest to ${LATEST}"
+          (git log origin/gh-pages..gh-pages | grep -q HEAD) && git push origin gh-pages
```

Comment:
Also here, I think we can just push, without any conditions, since the commits are only happening if the files changed. When we push without any new commits, git won't do anything.